pyrk/driver.py

```diff
@@ -106,15 +116,25 @@ def y0(si):
     :type si: SimInfo
     """
     i = 0
+    end_pg = 1 + si.n_pg
+    end_ref = 1 + si.n_pg + si.n_ref
+    end_dg = 1 + si.n_pg + si.n_ref + si.n_dg
     f = np.zeros(shape=(si.n_entries(),), dtype=float)
     f[i] = 1.0  # power is normalized is 1
     for j in range(0, si.n_pg):
         i += 1
         f[i] = f[0] * \
-            si.ne._pd.betas()[j]/(si.ne._pd.lambdas()[j]*si.ne._pd.Lambda())
+            si.ne._pd.betas()[j]/(si.ne._pd.lambdas()[j]*si.ne._Lambda)
+    assert(i == end_pg-1)
+    for l in range(0, si.n_ref):
+        i += 1
+        f[i] = f[0] * \
+            si.ne._ref_rho[l]/(si.ne._ref_lambda[l]*si.ne._Lambda)
+    assert(i == end_ref-1)
     for k in range(0, si.n_dg):
         i += 1
         f[i] = 0
+    assert(i == end_dg - 1)
```

Comment:
This function is now getting way too long to test. Additionally, it seems no new tests were added to reflect this new functionality. Dealing with this right now may be a bit cumbersome, as driver class needs to be refactored to avoid putting so much of this kind of logic in it. However, let's consider whether any of this (now massive) function can be moved out into smaller functions, either in the driver class or elsewhere.